packages/dnb-eufemia/src/extensions/forms/Field/Date/Date.tsx

```diff
@@ -208,6 +240,131 @@ function DateComponent(props: Props) {
   )
 }
 
+function parseRangeValue(value: DateProps['value']) {
+  return (
+    value
+      .split('|')
+      // Assign to null if falsy value, to properly clear input values
+      .map((value) => (/(undefined|null)/.test(value) ? null : value))
+  )
+}
+
+function validateDateLimit({
+  value,
+  isRange,
+  locale,
+  translations,
+  ...params
+}: {
+  value: DateProps['value']
+  minDate: DateProps['minDate']
+  maxDate: DateProps['maxDate']
+  isRange: DateProps['range']
+  locale: ProviderProps['locale']
+  translations: FormsTranslation['Date']
+}) {
+  if ((!params.minDate && !params.maxDate) || !value) {
+    return
+  }
+
+  const [startDateParsed, endDateParsed] = parseRangeValue(value)
+
+  const minDate = convertStringToDate(params.minDate)
+  const minDateISO = params.minDate
+    ? typeof params.minDate === 'string'
+      ? params.minDate
+      : params.minDate.toISOString()
+    : undefined
+
+  const maxDate = convertStringToDate(params.maxDate)
+  const maxDateISO = params.maxDate
+    ? typeof params.maxDate === 'string'
+      ? params.maxDate
+      : params.maxDate.toISOString()
+    : undefined
+
+  const startDate = convertStringToDate(startDateParsed)
+  const endDate = convertStringToDate(endDateParsed)
+
+  const options: FormatDateOptions = {
+    locale,
+    variant: 'long',
+  }
+
+  // Handle non range validation
+  if (!isRange) {
+    if (isBefore(startDate, minDate)) {
+      return new Error(
+        translations.errorMinDate.replace(
+          /%s/,
+          formatDate(minDateISO, options)
+        )
+      )
+    }
+
+    if (isAfter(startDate, maxDate)) {
+      return new Error(
+        translations.errorMaxDate.replace(
+          /%s/,
+          formatDate(maxDateISO, options)
+        )
+      )
+    }
+
+    return
+  }
+
+  const messages: Array<Error> = []
+
+  // Start date validation
+  if (isBefore(startDate, minDate)) {
+    messages.push(
+      new Error(
+        translations.errorRangeStartDateMinDate.replace(
+          /%s/,
+          formatDate(minDateISO, options)
+        )
+      )
+    )
+  }
+
+  if (isAfter(startDate, maxDate)) {
+    messages.push(
+      new Error(
+        translations.errorRangeStartDateMaxDate.replace(
+          /%s/,
+          formatDate(maxDateISO, options)
+        )
+      )
+    )
+  }
+
+  // End date validation
+  if (isBefore(endDate, minDate)) {
+    messages.push(
+      new Error(
+        translations.errorRangeEndDateMinDate.replace(
+          /%s/,
+          formatDate(minDateISO, options)
+        )
+      )
+    )
+  }
+
+  if (isAfter(endDate, maxDate)) {
+    messages.push(
+      new Error(
+        translations.errorRangeEndDateMaxDate.replace(
+          /%s/,
+          formatDate(maxDateISO, options)
+        )
+      )
```

Comment:
```suggestion
      new FormError('Date.errorRangeEndDateMaxDate', { messages: { myVar: formatDate(maxDateISO, options) } })
```

Something like this.